d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -534,34 +581,67 @@ private void handleResourceRemoval(Collection<String> removedResources, Resource
     {
       return;
     }
+
+    Map<String, ResourceSubscriber> subscribers = getResourceSubscriberMap(type);
+    WildcardResourceSubscriber wildcardSubscriber = getWildcardResourceSubscriber(type);
     for (String resourceName : removedResources)
     {
       _xdsClientJmx.incrementResourceNotFoundCount();
       _log.warn("Received response that {} {} was removed", type, resourceName);
-      ResourceSubscriber subscriber = getResourceSubscriberMap(type).get(resourceName);
+
+      ResourceSubscriber subscriber = subscribers.get(resourceName);
       if (subscriber != null)
       {
         subscriber.onRemoval();
       }
+
+      if (wildcardSubscriber != null)
+      {
+        wildcardSubscriber.onRemoval(resourceName);
+      }
     }
-    WildcardResourceSubscriber wildcardSubscriber = getWildcardResourceSubscriber(type);
-    if (wildcardSubscriber != null)
+  }
+
+  // If the nonce indicates that this is the last chunk of the response, notify the wildcard subscriber.
+  private void notifyOnLastChunk(DiscoveryResponseData response)
```

Comment:
Can you link the [diderot PR](https://github.com/linkedin/diderot/pull/3) that introduced this behavior?

Reply:
added